### PR TITLE
Add clipping, shapes and masking

### PR DESCRIPTION
**Pull request by jamesnw**
Targeting the [CSS4 "Clipping, shapes & masking"](https://github.com/CSS-Next/css-next/blob/main/list-of-css-features.md) group.

- [x] Clip-path (needed for #1624)
- [x] Shapes (needed for #1624)
- [x] Add group to Masks
- [x] Merge with masks-
  - [x] mask-composite
  - [x] mask-mode
- [x] New mask stories
  - [x] mask-border (different compat story, use case)
  - [x] mask-type (only applies to SVG <mask>, discrete feature)

### CODE REVIEW

File: groups/clipping-shapes-masking.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+name: Clipping, shapes and masking
+parent: css
```
Comment (captainbrosset) [inline groups/clipping-shapes-masking.yml:2]: The potential issue with this group is that it now contains features that apply to both CSS and SVG.
Your features make sense to me, and I'm fine with making them apply to both languages, if others are. But in that case, we need to make this group belong to both CSS and SVG (which is not possible today).

Comment (jamesnw) [reply groups/clipping-shapes-masking.yml:2]: Good point- I've been thinking we will need to address the overlap between CSS and SVG more thoroughly at some point. Allowing for multiple parents for a group would be complex, so I'll remove the parent group here for now.

File: features/path-shape.yml.dist
Comment (jamesnw) [inline features/path-shape.yml.dist:1]: This is a bit of a surprising result- these features do have some implementation, and I'm not sure if I'm misreading the BCD data, or if there is a bug in generating this.

**Comment by jamesnw** [comment]
I just saw #1129, which is blocked with more conversation in #1128.

File: features/clip-path-animatable.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+name: Animatable clipping paths
+description: Clipping paths in CSS are animated using transitions and CSS animations.
```
Comment (mirisuzanne) [inline features/clip-path-animatable.yml:2]:
```suggestion
description: Clipping paths in CSS are animated using CSS transitions and animations.
```

File: features/clip-path-animatable.yml [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+spec: https://drafts.fxtf.org/css-masking-1/#the-clip-path
```
Comment (mirisuzanne) [inline features/clip-path-animatable.yml:3]: The 'current' spec:
```suggestion
spec: https://www.w3.org/TR/css-masking/#the-clip-path
```

I would update the other spec links to match.

File: features/clip-path-boxes.yml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+name: Clip path boxes
+description: The `fill-box`, `stroke-box`, and `view-box` values for the `clip-path` set the part of the shape that will be used to clip the element.
```
Comment (mirisuzanne) [inline features/clip-path-boxes.yml:2]:
```suggestion
description: The `fill-box`, `stroke-box`, and `view-box` values for `clip-path` set an edge of the elements box to use as the clipping shape.
```

**Review by mirisuzanne — COMMENTED** [review]
I like this way of breaking things out to clarify the support for various parts. Left a few comments on things that I think could be clarified.

File: features/clip-path-animatable.yml (hunk 3 above)
Comment (jamesnw) [reply features/clip-path-animatable.yml:3]: I'll defer this until resolution of https://github.com/web-platform-dx/web-features/issues/1785

File: features/clip-path-boxes.yml (hunk 4 above)
Comment (jamesnw) [reply features/clip-path-boxes.yml:2]: I added an apostrophe in `element's` but wanted to verify that was the intent.

Comment (mirisuzanne) [reply features/clip-path-boxes.yml]: the intent was possessive, but I'm dyslexic. The element has a box. :)

Comment (jamesnw) [reply features/clip-path-boxes.yml]: I just wanted to confirm it wasn't like `attorneys general` or something 😆

**Review by HiddenCedarsInvestmentGroup — APPROVED** [review]
😊

**Review by HiddenCedarsInvestmentGroup — APPROVED** [review]
😊

File: features/path-shape.yml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+name: "`path()` shape"
```
Comment (ddbeck) [inline features/path-shape.yml:1]: Let's do this feature in a separate PR (from synchronous discussion)

File: features/clip-path-animatable.yml [hunk 6]
```diff
@@ -0,0 +1,6 @@
+name: Animatable clipping paths
+description: Clipping paths in CSS are animated using CSS transitions and animations.
```
Comment (ddbeck) [inline features/clip-path-animatable.yml:2]: Might be nice to make this more concrete:

```suggestion
description: The `clip-path` property can be animated using CSS transitions and animations.
```

File: features/clip-path.yml [hunk 7]
```diff
@@ -0,0 +1,19 @@
+name: clip-path
+description: The `clip-path` CSS property and SVG attribute sets an area of an element that should be displayed. Everything outside the area will be hidden.
```
Comment (ddbeck) [inline features/clip-path.yml:2]: For parallelism with the second sentence (and avoid "display" as a verb):

```suggestion
description: The `clip-path` CSS property and SVG attribute sets the visible area of an element. Everything outside the area will be hidden.
```

File: features/mask-type.yml [hunk 8]
```diff
@@ -0,0 +1,8 @@
+name: mask-type
+description: The `mask-type` CSS property on an SVG `<mask>` sets whether the mask is a `luminance` or `alpha` mask.
```
Comment (ddbeck) [inline features/mask-type.yml:2]:
```suggestion
description: The `mask-type` CSS property on an SVG `<mask>` element sets whether the mask is a `luminance` or `alpha` mask.
```

File: features/path-shape.yml (hunk 5 above)
Comment (ddbeck) [reply features/path-shape.yml:1]: To elaborate here: the BCD data is very messy. It has subfeatures to describe support in `d`, `shape-outside`, and `offset-path`, but _also_ represents that information as partial support on `path`. It ought to do one or the other. We'll need to fix this in BCD.

(Please @ me on that PR, when you open it.)

**Review by ddbeck — CHANGES_REQUESTED** [review]
A couple small requests and elaborations on previous requests

File: features/path-shape.yml (hunk 5 above)
Comment (jamesnw) [reply features/path-shape.yml]: Moved to https://github.com/web-platform-dx/web-features/pull/1799

File: features/clip-path.yml [hunk 9]
```diff
@@ -0,0 +1,19 @@
+name: clip-path
+description: The `clip-path` CSS property and SVG attribute sets the visible area of an element. Everything outside the area will be hidden.
```
Comment (ddbeck) [inline features/clip-path.yml:2]:
```suggestion
description: The `clip-path` CSS property and SVG attribute set the visible area of an element. Everything outside the area will be hidden.
```

**Review by ddbeck — APPROVED** [review]
There's a typo to fix, otherwise this is ready to go. Thank you!
